### DISCUSSION TITLE
Cowchain Farm: a Dapp built with Soroban and Flutter

### DISCUSSION THREAD

**Discussion by hasToDev** [stellar/sorobounty-spectacular#14] [General]
Hi,

I would like to share my contribution for Sorobounty Spectacular!

Cowchain Farm is a simple cow farm management game-like app where we can buy a cow, feed it three times a day to increase its value, and then sell it after three days of caring.

- Soroban Code: https://github.com/hasToDev/cowchain-farm-soroban
- Flutter Code: https://github.com/hasToDev/cowchain-farm-app
- Youtube Web Demo: https://youtu.be/e1sfjSI8wGo
- Youtube Code Walkthrough: https://youtu.be/8JdVxJfvWFc
- Tutorial: https://dev.to/hastodev/cowchain-farm-a-dapp-built-with-soroban-and-flutter-2me0
- Live Web App: https://cowchain.hasto.dev/

**Comment by vinamogit** [comment]
That's nice!

**Reply by hasToDev** [reply]
thanks @vinamogit, yours is awesome!

**Reply by ghost** [reply]
You all who participated created great dApps! Mine is very basic, but I'm happy to have learned a lot by myself, and most important, to have learned from all of your projects posted here, I will definitely keep experimenting with Soroban ❤️

**Reply by hasToDev** [reply]
Keep grinding @fernandodavidmartinez, we all are still learning here.

**Comment by briwylde08** [comment]
Hi there! Thank you so much for your submission to Sorobounty Spectacular! This is awesome. We'll review and get back to you within a week 🐄 

Best,
Bri

**Reply by hasToDev** [reply]
Thank you, Bri.

**Comment by vinamogit** [comment]
I'm playing it, and I think it could be a cool idle game.
I have some questioning about the login. You will get the cows with get_all_cows. This is an authenticated function, why?
Also, you actually execute the function while there seems to be no byte written, it could just be simulated, does there is a reason to pay to log in?

Nice choice of assets also.

**Reply by hasToDev** [reply]
Yeah, you're right. I overlooked the simulated transaction result and only focused on the transaction data.

Well, the idea is that only you who can read your cow data list.
Come to think of it, this approach is unnecessary since the data is public anyway.

Thank you for the input @vinamogit , nice catch, I'll make the improvement later.

**Comment by briwylde08** [comment]
Hello @hasToDev we've reviewed your submission! [🎉](https://emojipedia.org/party-popper/)

Given our guidelines, your submission may qualify for an award of $1,500 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Reply by hasToDev** [reply]
Hello Bri,
Thank you so much!

I have filled out the form.
I look forward to build another project with Soroban. 🚀

**Reply by briwylde08** [reply]
We love to hear it!!

**Comment by briwylde08** [comment]
Thank you for providing the requested information. You’ve passed KYC and we’ve logged your award payment of $1,500 worth of XLM, which will be sent to the wallet you designated in the next week.

Thanks again and please let me know if you have any questions!

Best,
Bri

**Comment by hasToDev** [comment]
Cowchain Farm has been updated to Soroban Preview 11. 🚀

**Comment by silence48** [comment]
Love it did not see it til today
